src/backend/aspen/api/middleware/session.py

```diff
@@ -0,0 +1,85 @@
+import hashlib
+
+from flask.json.tag import TaggedJSONSerializer
+from itsdangerous import URLSafeTimedSerializer
+from itsdangerous.exc import BadSignature
+from starlette.datastructures import MutableHeaders
+from starlette.requests import HTTPConnection
+from starlette.types import ASGIApp, Message, Receive, Scope, Send
+
+
+class SessionMiddleware:
```

Comment:
This middleware is heavily based on [starlette's session middleware](https://github.com/encode/starlette/blob/master/starlette/middleware/sessions.py), but with some small modifications to support [Flask's secure cookie implementation](https://github.com/pallets/flask/blob/main/src/flask/sessions.py#L316). This means that session cookies created by the flask app will be readable by the fastapi app, and vice versa.

Reply:
Looking at my browser's cookie for Staging and over the [Flask docs](https://flask.palletsprojects.com/en/2.0.x/api/?highlight=session#flask.session.permanent) for `session`, it looks like the auth cookie expires on browser `Session`. So it probably depends on our users usage, but I would assume for a good chunk of our users they're effectively permanent (I hadn't known until a few minutes ago that the "continue where I left off" options in browsers restores those Session cookies too. Now I know!).

I'm totally happy to keep the Flask session cookie approach, just hopefully we can move away from needing to keep the underlying Flask lib.